plugins/woocommerce/templates/myaccount/orders.php

```diff
@@ -39,12 +39,17 @@
 				$item_count = $order->get_item_count() - $order->get_item_count_refunded();
 				?>
 				<tr class="woocommerce-orders-table__row woocommerce-orders-table__row--status-<?php echo esc_attr( $order->get_status() ); ?> order">
-					<?php foreach ( wc_get_account_orders_columns() as $column_id => $column_name ) : ?>
-						<td class="woocommerce-orders-table__cell woocommerce-orders-table__cell-<?php echo esc_attr( $column_id ); ?>" data-title="<?php echo esc_attr( $column_name ); ?>">
+					<?php foreach ( wc_get_account_orders_columns() as $column_id => $column_name ) :
+						$is_order_number = 'order-number' === $column_id;
+						$cell_html_tag   = $is_order_number ? 'th scope="row"' : 'td';
+						$cell_html_attrs = 'class="woocommerce-orders-table__cell woocommerce-orders-table__cell-' . esc_attr( $column_id ) . '"' . ' ' . 'data-title="' . esc_attr( $column_name ) . '"';
+					?>
+						<?php echo '<' . $cell_html_tag . ' ' . $cell_html_attrs . '>'; ?>
```

Comment:
This works but is not ideal to directly echo this without any escaping. While it's safe right now as this value will always either be a `th` or `td`, if this code ever changes in the future, we may not be sure of that.

Wondering if instead we just have a conditional here that outputs the right thing, similar to the line added below: `<?php echo $is_order_number ? '</th>' : '</td>'; ?>`

Something like:
```php
<?php if ( $is_order_number ) : ?>
    <th scope="row" ATTRS HERE>
<?php else : ?>
    <td ATTRS HERE>
<?php endif; ?>
```

Reply:
Thanks for the feedback, @dkotter !
Please review the changes: [Refactor how order cell is rendered](https://github.com/woocommerce/woocommerce/pull/49201/commits/14cc1565edf28e3d75985db910039134654ebb03)